### PR TITLE
Add histogram computation capabilities to the grr cli

### PR DESCRIPTION
**Pull request by svetlin-mladenov**
Add a new command `histogram` to `grr_manage`. The command takes as arguments a genomic resource repo and a resource in that repo. It generates histograms for that resource. The histograms are described in the resource `genomic_resource.yaml` file. The output histogram is written in a directory `histograms` under the resource path.

The histogram calculation is done in parallel. The parallelism is based on the contigs - for each config a separate job. The number of jobs to be executed in parallel can be controlled through the `--jobs` command line parameter. The default behavior is to start as many jobs as processors on the machine.

### CODE REVIEW

File: dae/dae/genomic_resources/genomic_scores.py [hunk 1]
```diff
@@ -90,20 +91,51 @@ def _get_header(self):
...
 
     def get_all_chromosomes(self):
         return self.table.get_chromosomes()
 
     def get_all_scores(self):
         return list(self.score_columns)
 
+    def fetch_region(self, chrom: str, pos_begin: int, pos_end: int,
+                     scores: List[str]) \
+            -> dict[str, Generator[Number, None, None]]:
```
Comment (lchorbadjiev) [inline dae/dae/genomic_resources/genomic_scores.py:108]: I don't think this return type is a good idea. To have multiple independent generators would mean that we are going to iterate over all records multiple times.

I strongly prefer to have Generator[Dict[str, number], None, None].

File: dae/dae/genomic_resources/score_statistics.py [hunk 2]
```diff
@@ -127,6 +129,131 @@ def add_value(self, value):
...
+        limits = np.iinfo(np.int64)
+        scr_min, scr_max = limits.max, limits.min
+        for i_min, i_max in min_maxes:
+            scr_min = min(scr_min, i_min)
+            scr_max = max(scr_max, i_max)
+
+        logger.info(f"Done calculating min/max for {score_id}.\
+ min={scr_min} max={scr_max}")
+        assert scr_min <= scr_max
+        return scr_min, scr_max
```
Comment (lchorbadjiev) [inline dae/dae/genomic_resources/score_statistics.py:217]: consider re-organizing min/max calculation to become similar to histogram calculation _do_hist/_merge_hist

consider keeping the parallelization into one place - inside build() method.

**Review by lchorbadjiev — CHANGES_REQUESTED** [review]

File: dae/dae/genomic_resources/genomic_scores.py (hunk 1 above)
Comment (svetlin-mladenov) [reply dae/dae/genomic_resources/genomic_scores.py]: @lchorbadjiev fixed

File: dae/dae/genomic_resources/score_statistics.py (hunk 2 above)
Comment (svetlin-mladenov) [reply dae/dae/genomic_resources/score_statistics.py]: @lchorbadjiev fixed

**Review by lchorbadjiev — APPROVED** [review]

**Pull request merged**